### PR TITLE
feat(grid): add card view to grid

### PR DESCRIPTION
**Pull request by dkorobeinikov**

### CODE REVIEW

File: projects/playground/src/app/pages/grid/component/grid.component.scss [hunk 1]
```diff
@@ -4,3 +4,12 @@
     justify-content: center;
     align-items: center;
 }
+
+.card {
+    background: #ffffff;
+    border: 1px solid #cfd8dd;
```
Comment (toxik) [inline projects/playground/src/app/pages/grid/component/grid.component.scss:10]: apollo variables? :P

File: projects/playground/src/app/pages/grid/component/grid.component.html [hunk 2]
```diff
@@ -106,4 +107,18 @@ <h2>Expanded row with ID: {{ entry.id }}</h2>
             </div>
         </ng-template>
     </ui-grid-expanded-row>
+
+    <ui-grid-row-card-view>
```
Comment (toxik) [inline projects/playground/src/app/pages/grid/component/grid.component.html:111]: hmm.. is there a way maybe to reuse the actions row inside the card? we'll probably have the same actions

File: projects/playground/src/app/pages/grid/component/grid.component.scss (hunk 1 above)
Comment (dkorobeinikov) [reply projects/playground/src/app/pages/grid/component/grid.component.scss:10]: the repository doesn't have Apollo dependency and doesn't use any Apollo's variables

File: projects/playground/src/app/pages/grid/component/grid.component.html (hunk 2 above)
Comment (dkorobeinikov) [reply projects/playground/src/app/pages/grid/component/grid.component.html:111]: will take a look

File: projects/playground/src/app/pages/grid/component/grid.component.scss (hunk 1 above)
Comment (dkorobeinikov) [reply projects/playground/src/app/pages/grid/component/grid.component.scss:10]: also, it's a playground, not a component's code

**Comment by toxik** [comment]
Let's make the loading state consistent with the rows view please, by adding `pointer-events: none;` and the 50% opacity with its transition

**Comment by toxik** [comment]
Maybe we should also disable resizing the columns whilst in card view?

File: projects/angular/components/ui-grid/src/body/ui-grid-row-card-view.directive.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import {
+    ContentChild,
+ Directive, TemplateRef,
+} from '@angular/core';
+
+export interface IGridRowCardViewContext {
```
Comment (toxik) [inline projects/angular/components/ui-grid/src/body/ui-grid-row-card-view.directive.ts:6]:
```suggestion
export interface IGridRowCardViewContext<T> {
```

File: projects/angular/components/ui-grid/src/body/ui-grid-row-card-view.directive.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    index: number;
+    last: boolean;
+    data: any;
```
Comment (toxik) [inline projects/angular/components/ui-grid/src/body/ui-grid-row-card-view.directive.ts:9]:
```suggestion
    data: T;
```

File: projects/angular/components/ui-grid/src/body/ui-grid-row-card-view.directive.ts [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+}
+
+@Directive({ selector: '[uiGridRowCardView], ui-grid-row-card-view' })
+export class UiGridRowCardViewDirective {
```
Comment (toxik) [inline projects/angular/components/ui-grid/src/body/ui-grid-row-card-view.directive.ts:13]:
```suggestion
export class UiGridRowCardViewDirective<T> {
```

File: projects/angular/components/ui-grid/src/body/ui-grid-row-card-view.directive.ts [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+    @ContentChild(TemplateRef, {
+        static: true,
+    })
+    html?: TemplateRef<IGridRowCardViewContext>;
```
Comment (toxik) [inline projects/angular/components/ui-grid/src/body/ui-grid-row-card-view.directive.ts:17]:
```suggestion
    html?: TemplateRef<IGridRowCardViewContext<T>>;
```

**Review by toxik — COMMENTED** [review]
would this help?

**Comment by dkorobeinikov** [comment]
> 

done

File: projects/angular/components/ui-grid/src/ui-grid.component.html [hunk 7]
```diff
@@ -401,6 +435,59 @@
...
+</ng-template>
+
+<ng-template #defaultCardTemplate
+    let-row="data"
+    let-index="index">
+    <div class="ui-grid-card-default">
+        <ng-container *ngFor="let column of renderedColumns$ | async">
+            <div [class.ui-grid-primary]="column.directive.primary"
+                 [class.ui-grid-secondary]="!column.directive.primary"
+                 [class.ui-grid-state-resizing]="column.directive === resizeManager.current?.column"
```
Comment (toxik) [inline projects/angular/components/ui-grid/src/ui-grid.component.html:469]: don't think we need this, as the cards dimensions are not controlled by the column widths
```suggestion
```

**Review by toxik — APPROVED** [review]

**Pull request merged**